libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/app/MobileSyncUpgradeManager.kt

```diff
@@ -83,7 +61,29 @@ protected synchronized void upgradeSObject() {
      *
      * @return Currently installed version of mobile sync.
      */
-    public String getInstalledSobjectVersion() {
-        return getInstalledVersion(MOBILE_SYNC_KEY);
+    val installedSobjectVersion: String
+        get() = getInstalledVersion(MOBILE_SYNC_KEY)
+
+    companion object {
+        /**
+         * Key in shared preference file for mobile sync version.
+         */
+        private const val MOBILE_SYNC_KEY = "mobile_sync_version"
+        private const val TAG = "MobileSyncUpgradeManager"
+        private var INSTANCE: MobileSyncUpgradeManager? = null
+
+        /**
+         * Returns an instance of this class.
+         *
+         * @return Instance of this class.
+         */
+        @get:Synchronized
+        val instance: MobileSyncUpgradeManager
+            get() {
+                if (INSTANCE == null) {
+                    INSTANCE = MobileSyncUpgradeManager()
+                }
+                return INSTANCE!!
```

Comment:
This works, though I'm always curious when I see a force unwrap and if control flow in Kotlin.  It looks like `INSTANCE` is private, so this could use a thread-safe lazy delegate like this, if you want.

```
val instance by lazy { MobileSyncUpgradeManager() }
```

I checked to see if it compiles that way, at least in MSDK's own workspace.

Reply:
I believe that lets us take the get:Synchronized away as well.